[Yin Ruohang] iP

yin72257 opened the pull request:
# My Duke Project

> This Duke Project is my favorite - Albert Einstein

**9/10 Dentists recommend this Duke Project**😱😱😱😱

* Complete with UI
* Very _cool_
* Has big features

Steps to add to this project
1. Clone to your repo
2. Add whatever you want
3. Pull request and I'll check it out

Features
- [x] GUI
- [x] Levels 1-10
- [ ] Level 11 onwards (coming soon)

Here's a code snippet from `Duke.java`
```java
Duke(String filePath) {
    storage = new Storage(filePath);
    parser = new Parser();
    try {
        tasks = new TaskList(storage.load());
        parser.addTaskList(tasks);
        parser.addStorage(storage);
    } catch (DukeException e) {
        tasks = new TaskList();
    }
}
```

~~[Don't click this link](https://github.com/nus-cs2103-AY2122S1/ip/pull/275)~~

Commit:
Add Gradle support

yin72257 committed:
Completed Duke Level 1

yin72257 committed:
Finished Level-2

yin72257 committed:
Level-3 Added

yin72257 committed:
Finished Level-4 with inheritance

yin72257 committed:
changed if else to switch

yin72257 committed:
added testing and Level-6

yin72257 committed:
fixed bat testing

yin72257 committed:
Complete Level 7

yin72257 committed:
Basic date functionality

yin72257 committed:
Merge branch 'branch-Level-7'

yin72257 committed:
Merge branch 'branch-Level-8'

yin72257 committed:
Add storage and task list classes

yin72257 committed:
Add A-MoreOOP

yin72257 committed:
Put everything in package duke

yin72257 committed:
Add JUnit testing

yin72257 committed:
Compile jar

yin72257 committed:
Javadocs annotations

yin72257 committed:
Fixed a few styling issues

yin72257 committed:
Completed Level 9

yin72257 committed:
Merge branch 'branch-Level-9'

yin72257 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

yin72257 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Todo.java
#	src/main/java/duke/Ui.java

lwlshawn reviewed (commented):
The only small nit i have is with Parser; since the course code quality standard recommends having shorter methods. It might make it easier to maintain if you shifted the error handling and printing responsibilities to other classes. Otherwise, code looks great :)

yin72257 committed:
Merge branch 'add-gradle-support'

yin72257 committed:
Added gradle support with run and build working

yin72257 committed:
Add Checkstyle for gradle and fix detected style issues

yin72257 committed:
implement basic functioning Javafx gui element

bklimey reviewed (commented):
Your code is well-documented and is pleasant to read. Variable names are descriptive and easy to understand. Overall, I did not find any coding standard violations. Well done and keep it up! 😁 👍

NicolasCwy commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,30 @@
+    private Duke duke = new Duke("data/duke.txt");
+
+    @Override
+    public void start(Stage stage) {
```
Saw this in a few places, remember to add header comments for all public methods!

NicolasCwy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+            }
+            String[] messageArr;
+            switch (inputArr[0]) {
+            case "todo":
```
Good indentation for the switch case!

NicolasCwy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,55 @@
+     *
+     * @return "X" for done and " " for incomplete.
+     */
+    public String getStatusIcon() {
```
Nice use of descriptive method names 💯

NicolasCwy reviewed (commented):
Congratulations on your first PR!🥳 You code is well written and follows the coding standards. You could consider adding javadocs to the methods you have just added, checkstyle would help you easily identify them and is relatively easy to set-up. ATB!

lwlshawn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+     * @param input Input string by user
+     * @return True or false. If false, program ends. If true, program continues.
+     */
+    public String parse(String input) {
```
Sorry for commenting on this again! Same point as before but 2103 requires review comments

lwlshawn reviewed (commented):
for 2103 script

yin72257 committed:
Add assertions to ensure parser loaded properly.

As parse() function is a public function, it may be called without correct states being initialized.

Let's add assertions to make sure that tasklist and storage is initalized before parsing any input.

Assertions are not part of the main code but can be used to make sure certain things are running correctly.
It also runs fast and is fast to implement.

yin72257 committed:
Refactor parser code to be in their own function.

To maintain good code quality by following the SLAP principle, the logic in the parse function should be abstracted out to private functions.

Isolate logic of handling different commands to private functions.

Allows the reader to understand the logic better.

yin72257 committed:
Merge pull request #2 from yin72257/branch-A-Assertions

Add assertions to ensure parser loaded properly.

yin72257 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Parser.java

yin72257 committed:
Merge pull request #1 from yin72257/branch-A-CodeQuality

Refactor parser code to be in their own function.

yin72257 committed:
Add task feature.
Overhaul save format

yin72257 committed:
Merge branch 'branch-C-Tagging'

yin72257 committed:
Add more JUnit test

yin72257 committed:
Bug fixes with save todo.
Style fixes

yin72257 committed:
Add title and Ui.png

yin72257 committed:
Change find display message

yin72257 committed:
Add basic UG

yin72257 committed:
Move Ui

yin72257 committed:
Update readme and Ui.png

yin72257 committed:
Fix User guide markdown
